homeassistant/components/switch/broadlink.py

```diff
@@ -54,7 +54,25 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     try:
         broadlink_device.auth()
     except socket.timeout:
-        pass
+        _LOGGER.error("Failed to connect to device.")
+
+    def _learn_command(call):
+        try:
+            broadlink_device.auth()
+        except socket.timeout:
+            _LOGGER.error("Failed to connect to device.")
+            return
+        broadlink_device.enter_learning()
+        _LOGGER.info("Press the key you want HASS to learn")
+        start_time = utcnow()
+        while (utcnow() - start_time) < timedelta(seconds=20):
```

Comment:
So right now the current code doesn't seem to be very async friendly. How long does check_data takes ? Is that something that blocks for a second? If that returns instantly, this process will take up a lot of CPU as this loop will run for 20 seconds straight !

If you want to do it async:
 - mark `_learn_command` as `asyncio.coroutine`
 - Run all methods on broadlink_device in the executor
 - `yield from asyncio.sleep(1, loop=hass.loop)` to sleep

btw would it make sense to instead of logging the found packet, to add it as a persistent notification ?